blogs/contractionorder.md

```diff
@@ -0,0 +1,338 @@
+In the following sections, we will introduce a method to find the optimal contraction order based on the optimal tree decomposition.
+
+## Finding the Optimal Contraction Order via Tree Decomposition
+
+In the previous section, we introduce the concept of tensor network and its contraction order, so that now you should understand why the contraction order so important. 
+Then the next question is how to find the optimal contraction order.
+
+In this section, we will introduce one way to find the theoretical optimal contraction order based on the exact tree width solver, according to the following well known theorem[^Markov] :
+
+**Theorem 1**. Let $C$ be a quantum circuit with $T$ gates and whose underlying circuit graph is $G_c$. Then $C$ can be simulated deterministically in time $T^{O(1)} e^{O(tw(G_C))}$, where $tw(G_C)$ is the treewidth of $G_C$.
+
+Using the language of tensor network, we can rewrite the above theorem as follows: the bottleneck of time complexity of the contraction of a tensor network is $O(e^{O(tw(L(G)))})$, where $L(G)$ is the line graph of the hypergraph representation of the tensor network. 
+Therefore, if we can find the tree decomposition of the tensor network with minimal treewidth, we can find the optimal contraction order of the tensor network.
+
+To further make use of the theorem, we need to introduce the concepts of line graph, tree width and tree decomposition.
+
+### Line Graph
+
+A formal definition of the [**line graph**](https://en.wikipedia.org/wiki/Line_graph) is as follows:
+
+**Definition 1**. Given a graph G, its line graph $L(G)$ is a graph such that: 
+   1. each vertex of $L(G)$ represents an edge of $G$; 
+   2. two vertices of $L(G)$ are adjacent if and only if their corresponding edges share a common endpoint ("are incident") in $G$. 
+
+That is, it is the intersection graph of the edges of G, representing each edge by the set of its two endpoints.
+
+For a tensor network, we can construct a hypergraph $G$ whose vertices are the tensors and whose hyperedges are the indices. 
+Then the line graph $L(G)$ of the hypergraph $G$ is the graph whose vertices are the indices and whose edges are the tensors, which is a simple graph, as shown in the following figure:
+
+![Fig.1](/assets/treewidth_figs/linegraph.png)
+
+Since we are considering a tensor network, dimension of the indices have to be considered. 
+Therefore, for each vertex of the line graph $L(G)$, we define its weight as $\log_2(d)$, where $d$ is the dimension of the index. 
+
+In this way, any possible tensor can be represented as a subgraph induced by a set of vertices in $L(G)$, and its log2 size is given by the sum of weights of the boundary vertices of the subgraph. For example, the tensor $C$ in the figure above corresponding to the subgraph induced by the vertices $\{b, d, f\}$, and its log2 size is 
+$${dims(b) \times dims(d) \times dims(f)} = 2^{w(b) + w(d) + w(f)}$$
+where $w(*)$ for weights of vertices.
```

Comment:
I am not sure what tensor $C$ in the figure above is referring to, and in particular b, d and f. Tensor $C$ in the figure above has indices $j$, $k$ and $n$. But is it the goal to just use a single original tensor as example, or did you intend to refer to an already partially contracted network.

Reply:
Thank you very much for your suggestions. I agree with you that here we say a tensor $C$ corresponding to a subgraph in $L(G)$ is not correct, I removed these statements.